include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -398,6 +450,10 @@ class zip_iterator
         return !(*this < __it);
     }
 
+    friend auto
+    is_onedpl_indirectly_device_accessible_iterator(const zip_iterator&)
```

Comment:
Just a couple of questions.
Do we have some risk that we use not fully qualified name for `zip_iterator`?
Do we have some risk that something  `zip_iterator` value (with qualificators) doesn't
match to `const zip_iterator&`?